From c2f4ff8d26d2d7510e32f8ec6bcaa222af95ed9a Mon Sep 17 00:00:00 2001
From: Eric Garver <eric@garver.life>
Date: Tue, 14 Oct 2025 17:50:22 -0400
Subject: [PATCH 41/48] v2.4.0: test(ipset): scale: verify all the entries were
 added

(cherry picked from commit ab35fc7a99536efc7dde2242507fad6bed577d68)
---
 src/tests/regression/ipset_scale.at | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/tests/regression/ipset_scale.at b/src/tests/regression/ipset_scale.at
index 0aef986434f0..d754e035f187 100644
--- a/src/tests/regression/ipset_scale.at
+++ b/src/tests/regression/ipset_scale.at
@@ -1,6 +1,8 @@
 FWD_START_TEST([ipset scale])
 AT_KEYWORDS(ipset gh738 scale)
 
+AT_SKIP_IF([! NS_CMD([which wc >/dev/null 2>&1])])
+
 dnl Create a huge ipset
 AT_CHECK([touch ./entries], 0, [ignore])
 AT_CHECK([sh -c '
@@ -22,4 +24,11 @@ ulimit -d $(expr 1024 \* 300)
 FWD_RESTART() dnl required because we changed ulimit
 FWD_RELOAD()
 
+dnl Verify all the entries are added.
+m4_if(nftables, FIREWALL_BACKEND, [
+    WAIT_UNTIL([test "$(NS_CMD([nft $NFT_NUMERIC_ARGS list set inet firewalld foobar |wc -l]))" -eq 31256])
+], [
+    WAIT_UNTIL([test "$(NS_CMD([$IPSET list foobar |wc -l]))" -eq 62508])
+])
+
 FWD_END_TEST()
-- 
2.47.3

